src/Api/Response/Parser/FiltersParser.php

```diff
@@ -39,9 +39,12 @@ public function __construct(LibraryCallContract $libraryCallContract, Image $col
         $this->imageService = $colorImageService;
     }
 
-    public function parse(SimpleXMLElement $data): array
+    /**
+     * @param SimpleXMLElement|null $data
+     */
+    public function parse($data): array
     {
-        if (!$data) {
+        if (!$data instanceof SimpleXMLElement) {
```

Comment:
Shouldn't this be typehinted like `parseForWidgets` instead? They receive the same data anyway.

Reply:
If one isn't of type `SimpleXMLElement` the other wouldn't be of type `SimpleXMLElement` either. Either you handle both like that, or you add a typehint to both. Not sure if there may be a case where `$data` wouldn't be of `SimpleXMLElement`, but since you added this check, you probably have more insight into this.

Reply:
Not sure about such cases but there was a test for that, which was actually failing because this method couldn't handle its input. Rather than adjusting the test, I adjusted the code, seemed safer. Matched the typehints of both methods.